source/meta.tex

```diff
@@ -5368,6 +5495,118 @@
 A reflection of the function designated by \tcode{fn}.
 \end{itemdescr}
 
+%FIXME: inconsistent title case;
+%I'm pretty sure we normally use sentence case.
+%Also, this title is duplicated and weird.
+%Why isn't this called "array reflection" in the style of the subclause above?
+\rSec2[meta.reflection.array]{Promoting to Static Storage}
+
+%FIXME: We generally don't like "compile-time".
+%This sentence is pretty nonsensical because these terms aren't mutually exclusive.
+%
+%We also deal with ranges as input, and ranges aren't necessarily backed by
+%storage (except for an empty class).
+%
+%Maybe we could say something along the lines of converting ranges of structural types
+%into arrays, during program translation.
+\pnum
+The functions in this subclause promote compile-time storage into static storage.
+
+\indexlibraryglobal{reflect_constant_string}%
+\begin{itemdecl}
+template<ranges::@\libconcept{input_range}@ R>
+  consteval info reflect_constant_string(R&& r);
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+Let \tcode{CharT} be \tcode{ranges::range_value_t<R>}.
+
+\pnum
+\mandates
+\tcode{CharT} is one of
+%FIXME: this is kinda silly; we have the defined term "character type"
+%which matches this set exactly.
+%We may as well use it.
+\tcode{char},
+\tcode{wchar_t},
+\tcode{char8_t},
+\tcode{char16_t},
+\tcode{char32_t}.
+
+\pnum
+Let $V$ be the pack of values of type \tcode{CharT}
+whose elements are the corresponding elements of \tcode{r},
+except that if \tcode{r} refers to a string literal object,
+then $V$ does not include the trailing null terminator of \tcode{r}.
+
+\pnum
+Let $P$ be the template parameter object\iref{temp.param}
+of type \tcode{const CharT[sizeof...(V)+1]}
+initialized with \tcode{$V$..., CHART()}.
+
+\pnum
+\returns
+\tcode{\reflexpr{$P$}}.
+
+\pnum
+\begin{note}
+$P$ is a potentially non-unique object\iref{intro.object}.
+\end{note}
+\end{itemdescr}
+
+\indexlibraryglobal{reflect_constant_array}%
+\begin{itemdecl}
+template<ranges::@\libconcept{input_range}@ R>
+  consteval info reflect_constant_array(R&& r);
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+Let \tcode{T} be \tcode{ranges::range_value_t<R>}.
+
+\pnum
+\mandates
+\tcode{T} is a structural type\iref{temp.param},
+\tcode{is_constructible_v<T, ranges::range_reference_t<R>>} is \tcode{true}, and
+\tcode{is_copy_constructible_v<T>} is \tcode{true}.
+
+\pnum
+\constantwhen
+\tcode{reflect_constant(e)} is a constant subexpression
+for every element \tcode{e} of \tcode{r}.
+
+\pnum
+Let $V$ be the pack of values of type \tcode{info}
+of the same size as \tcode{r},
+where the $i^\text{th}$ element is \tcode{reflect_constant($\tcode{e}_i$)},
+where $\tcode{e}_i$ is the $i^\text{th}$ element of \tcode{r}.
+
+\pnum
+Let $P$ be
+\begin{itemize}
+\item
+  If \tcode{sizeof...($V$) > 0},
+  %FIXME: is true
```

Comment:
Or use math font or say "is greater than zero" and then we don't need the clunky "is `true`".

Reply:
Meh, it may be clunky, but it's also fairly short and familiar.